### DISCUSSION TITLE
It's not working!

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#3352] [Q&A] [answered]
I'm trying to add syntax highlighting for a new language.

```
# single comments, like # in this file
comment_single=#
# multiline comments
comment_open=#~
comment_close=~#
```

The above is not working. For a single line comment, it will only highlight the first word but not the whole line, e.g: `# foo bar` then it will only highlight `# foo` but not `bar` (update: this is also true for multiline comment). For multiline comment, the closing block `~#` is not highlighted.

I copied filetypes.csharp and modify it to create my own filetypes.foo.conf. I have the impression that it's using C's lexer so it treat these not as comments but preprocessor statements. Does what I wanted to do could be done only on the filetypes.foo.conf or I need to dive into Geany's source code?

**Comment by elextr** [comment, marked as answer]
These settings have nothing to do with highlighting.  Highlighting is determined by the Lexilla lexer selected by the filetype.  See the manual for making a custom filetype based on an existing filetype lexer if your language is close enough, otherwise you will need to create a new Lexer, see the Lexilla project.

**Reply by ghost** [reply]
Then I will have to dive into C++? The language is almost the same as C#, the only different is it use different way to comments.

**Reply by elextr** [reply]
Well try the C# filetype and see if its "good enough" otherwise you indeed will need to adjust the lexer code.

**Reply by ghost** [reply]
The way to comment is different to C language family: # for single line comment and #~ for opening and ~# for closing multiline comment.

**Reply by elextr** [reply]
Well then you will have to make a new lexer by changing the nearest one.  Lexers are for identifying lexical entities in a specific programming language and since comment formats don't change often in an established language AFAIK all lexers hard code the comment formats for their language.
